pkgs/development/python-modules/yq/default.nix

```diff
@@ -22,8 +22,13 @@ buildPythonPackage rec {
     sha256 = "1q4rky0a6n4izmq7slb91a54g8swry1xrbfqxwc8lkd3hhvlxxkl";
   };
 
+  patches = [ ./jq-path.patch ];
+
   postPatch = ''
-    substituteInPlace test/test.py --replace "expect_exit_codes={0} if sys.stdin.isatty() else {2}" "expect_exit_codes={0}"
+      substituteInPlace yq/__init__.py --subst-var-by JQ "${lib.getBin pkgs.jq}/bin/jq"
+      substituteInPlace test/test.py \
+          --subst-var-by JQ "${lib.getBin pkgs.jq}/bin/jq" \
+          --replace "expect_exit_codes={0} if sys.stdin.isatty() else {2}" "expect_exit_codes={0}"
   '';
```

Comment:
You can actually replace the variables you have with substitute within the `patches` declaration so you don't really need the postPatch for the JQ substitution.